### PR TITLE
Application completion

### LINKED ISSUE #15: User Stories 6-7

### PR DESCRIPTION
**Pull request by noahdurbin**
- user stories 6 and 7
- submit application button
- logic for submit button and add pets section
- testing for above functionality

### CODE REVIEW

File: app/views/applications/show.html.erb [hunk 1]
```diff
@@ -20,3 +20,13 @@
     <% end %>
   <% end %>
 <% end %>
+
+<% if @application.pets.count > 0 && @application.status == "In Progress"%>
+  <%= "Submit Application" %>
+  <%= form_with url: "/applications/#{@application.id}?status=Pending", method: :patch, params: {application_id: @application.id}, local: true do |form| %>
```
Comment (JCL461437) [inline app/views/applications/show.html.erb:26]: Nice knowing that we can use query a certain param we want to send to the Controller.

File: config/routes.rb [hunk 2]
```diff
@@ -4,9 +4,10 @@
   get '/applications/new', to: 'applications#new'
   get '/applications/:id', to: 'applications#show'
   post '/applications', to: 'applications#create'
+  patch '/applications/:id', to: 'applications#update'
 
   get '/pet_applications/new', to: 'pet_applications#new'
-  post 'applications/:application_id/pet_applications', to: 'pet_applications#create'
+  post '/applications/:application_id/pet_applications', to: 'pet_applications#create'
```
Comment (JCL461437) [inline config/routes.rb:10]: Nice work with the paths. Maybe we will end of re-organizing them later like Kat mentioned

File: spec/models/application_spec.rb [hunk 3]
```diff
@@ -1 +1,33 @@
...
+    @shelter_2 = Shelter.create(name: 'RGV animal shelter', city: 'Harlingen, TX', foster_program: false, rank: 5)
+
+    @pet_1 = @shelter_1.pets.create(name: 'Mr. Pirate', breed: 'tuxedo shorthair', age: 5, adoptable: false)
+    @pet_2 = @shelter_1.pets.create(name: 'Clawdia', breed: 'shorthair', age: 3, adoptable: true)
+
+    @application1 = Application.create!(name: 'John Doe', address: '123 Elm St', city: 'Denver', state: 'CO',
+                                        zip: '12345', description: 'Looking for a friendly dog', status: 'In Progress')
+  end
+
+  describe 'class methods' do
```
Comment (JCL461437) [inline spec/models/application_spec.rb:29]: I'll have to flush out these model tests once we write more methods in the models.

File: spec/models/pet_application_spec.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    @shelter_2 = Shelter.create(name: 'RGV animal shelter', city: 'Harlingen, TX', foster_program: false, rank: 5)
+
+    @pet_1 = @shelter_1.pets.create(name: 'Mr. Pirate', breed: 'tuxedo shorthair', age: 5, adoptable: false)
+    @pet_2 = @shelter_1.pets.create(name: 'Clawdia', breed: 'shorthair', age: 3, adoptable: true)
+
+    @application1 = Application.create!(name: 'John Doe', address: '123 Elm St', city: 'Denver', state: 'CO',
+                                        zip: '12345', description: 'Looking for a friendly dog', status: 'In Progress')
+  end
+
+  describe 'class methods ' do
```
Comment (JCL461437) [inline spec/models/pet_application_spec.rb:25]: I will add to the class method tests once we actually have more class methods written for the models.

File: spec/features/applications/show_spec.rb [hunk 5]
```diff
@@ -53,5 +53,31 @@
         expect(page).to have_content('All Pets: Buddy')
       end
     end
+
```
Comment (JCL461437) [inline spec/features/applications/show_spec.rb:56]: Great work with the tests Noah!

File: app/controllers/applications_controller.rb [hunk 6]
```diff
@@ -21,4 +19,20 @@ def create
       flash[:alert] = "Error: #{error_message(@application.errors)}"
     end
   end
+
+  def update
```
Comment (JCL461437) [inline app/controllers/applications_controller.rb:23]: Good to know the discrepancy between update and create, great application of update here.

**Review by JCL461437 — APPROVED** [review]

**Review by AuroraTuring — APPROVED** [review]

**Pull request merged**